YamlSplitter/FragmentModelHelper.cs

```diff
@@ -0,0 +1,180 @@
+                return null;
+            }
+            errorHappened = 0;
+            var filesByUid = new Dictionary<string, FragmentFile>();
+            foreach (var fragmentFile in fileBag)
+            {
+                foreach (var fragmentPair in fragmentFile.FragmentsByUid)
+                {
+                    if (filesByUid.ContainsKey(fragmentPair.Key))
+                    {
+                        Console.WriteLine("Dup Uid found in below two files:");
+                        Console.WriteLine(fragmentFile.FilePath);
+                        Console.WriteLine(filesByUid[fragmentPair.Key].FilePath);
+                        errorHappened++;
+                    }
+                    else
+                    {
+                        filesByUid.Add(fragmentPair.Key, fragmentFile);
+                    }
+                }
+            }
+            if (errorHappened > 0)
+            {
+                Console.WriteLine("Dup Uid found in {0} files, aborting...", errorHappened);
+                return null;
+            }
+            return filesByUid;
+        }
+
+        public static Dictionary<string, MarkdownFragment> LoadMarkdownFragment(string fileName)
+        {
+            if (string.IsNullOrEmpty(fileName))
+            {
+                return null;
+            }
+            if (!File.Exists(fileName))
+            {
+                return new Dictionary<string, MarkdownFragment>();
+            }
+            var markdown = File.ReadAllText(fileName);
+            var ast = Markdown.Parse(markdown);
+            var models = new MarkdownFragmentsCreater().Create(ast).ToList();
+            return models.ToDictionary(m => m.Uid, m => m.ToMarkdownFragment(markdown));
+        }
+
+        public static MarkdownFragment ToMarkdownFragment(this MarkdownFragmentModel model, string originalContent)
+        {
+            Dictionary<string, object> metadata = null;
+            if (!string.IsNullOrEmpty(model.YamlCodeBlock))
+            {
+                using (TextReader sr = new StringReader(model.YamlCodeBlock))
+                {
+                    metadata = Common.YamlUtility.Deserialize<Dictionary<string, object>>(sr);
+                }
+            }
+
+            return new MarkdownFragment()
+            {
+                Uid = model.Uid,
+                Metadata = metadata,
+                Properties = model.Contents?.Select(prop => prop.ToMarkdownProperty(originalContent)).ToDictionary(p => p.OPath, p => p)
+            };
+        }
+
+        public static MarkdownProperty ToMarkdownProperty(this MarkdownPropertyModel model, string originalContent)
+        {
+            var content = "";
+            if (model.PropertyValue?.Count > 0)
+            {
+                var start = model.PropertyValue.First().Span.Start;
+                var length = model.PropertyValue.Last().Span.End - start + 1;
+                var piece = originalContent.Substring(start, length);
```

Comment:
Should `MarkdownPropertyModel` contain original content as `Content` so that there is no need to do calculation here? @superyyrrzz @928PJY

Reply:
@vicancy MarkdownPropertyModel is parsed from ast, which only contains the start/end info.